### PR TITLE
Add Rc2 releases changes

### PR DESCRIPTION
**Pull request by PaulHigin**
This PR contains changes for RC2 release, including bug fixes and new support for secret metadata.
Documents and help have been updated.

### CODE REVIEW

**Comment by mi-hol** [comment]
@SydneyhSmith should this PR not be merged to correct the published documentation?

File: Docs/DesignDoc.md [hunk 1]
```diff
@@ -48,6 +48,24 @@ Secret objects supported by this module are currently limited to:
 
 - Hashtable - Hash table of name value pairs, where values are restricted to the above secret types.  
 
+## Secret metadata
+
+Extension vaults can optionally support storing and retrieving additional secret metadata, that is data associated with the secret.  
+Secret metadata is not security sensitive and does not need to be stored securely in the extension vault.  
+
+Secret metadata can be included by using the `-Metadata` parameter: `Secret-Secret -Metadata @{ Name=Value }`.
```
Comment (anamnavi) [inline Docs/DesignDoc.md:56]: do you mean `Set-Secret -Metadata @{Name=Value}`?

File: README.md [hunk 2]
```diff
@@ -32,6 +32,24 @@ You can install the `Microsoft.PowerShell.SecretManagement` module from the [Pow
 Install-Module -Name Microsoft.PowerShell.SecretManagement -Repository PSGallery
 ```
 
+## Secret metadata
+
+Extension vaults can optionally support storing and retrieving additional secret metadata, that is data associated with the secret.  
+Secret metadata is not security sensitive and does not need to be stored securely in the extension vault.  
+
+Secret metadata can be included by using the `-Metadata` parameter: `Secret-Secret -Metadata @{ Name=Value }`.
```
Comment (anamnavi) [inline README.md:40]: do you mean Set-Secret -Metadata @{Name=Value}?

File: Docs/DesignDoc.md (hunk 1 above)
Comment (PaulHigin) [reply Docs/DesignDoc.md:56]: Yes, thanks!

File: src/code/Utils.cs [hunk 3]
```diff
@@ -21,6 +22,13 @@ internal static class Utils
     {
         #region Members
 
+        private const string NoVaultRegistered = @"
+        There are currently no extension vaults registered.
+        At least one vault must be registered before SecretManagement can add or retrieve secrets.
+        You can download SecretManagement extension vault modules from PowerShellGallery.
+        https://aka.ms/SMVaults
```
Comment (SteveL-MSFT) [inline src/code/Utils.cs:29]: I would prefer we spelled out: https://aka.ms/SecretManagementVaults since folks are likely to cut and paste it anyways.

**Review by SteveL-MSFT — APPROVED** [review]
LGTM, one minor nit

**Review by anamnavi — APPROVED** [review]
LGTM too

**Pull request merged**